### ISSUE TITLE
Decide on a classifier interface for ML3 to use

### ISSUE THREAD

**Issue by florianmai** [Modalities/ml_filter#89]
Since our classifier has a custom classifier head that makes it efficient to predict multiple scores at once, we need to define an interface so that it can be easily used during inference (issue #22).

If I remember correctly, ML3 has been assuming the XLMRobertaForSequenceClassification interface. @timpal0l is this correct?

My suggestion is to extend the XLMRobertaForSequenceClassification in such a way that it incorporates our changes to the architecture. [I tried this out in this commit.](https://github.com/EuroLingua-GPT/ml_filter/commit/63d9da49c614cd8e48623d770e747054bb5922db) After training, the model can then again be loaded in a simple way like shown below. The shape of the output (which are currently logits) depends on whether it's using regression or not. We can adjust it as needed. @timpal0l would this suit your needs?

```Python3
from ml_filter.utils.train_classifier import XLMRobertaForMultiTargetClassification

# Load the model
trained_model = XLMRobertaForMultiTargetClassification.from_pretrained(
    "/home/fmai/ml_filter/data/output/annotated_fineweb_model_test/final",  # This should be the output_dir you specified during training
    num_metrics=4,  # Use the same parameters as during training
    num_classes_per_metric=torch.tensor([6, 6, 6, 2]),  # Example values
    regression=False
)

# Load tokenizer
from transformers import AutoTokenizer
tokenizer = AutoTokenizer.from_pretrained('FacebookAI/xlm-roberta-base')

# Test if it works
trained_model.eval()
with torch.no_grad():
    # Create a sample input
    test_text = "This is a test document"
    inputs = tokenizer(test_text, return_tensors="pt", truncation=True, max_length=512)
    
    # Get predictions
    outputs = trained_model(**inputs)
    predictions = outputs.logits
    print(predictions)
```
